specifyweb/frontend/js_src/lib/components/QueryBuilder/Wrapped.tsx

```diff
@@ -537,6 +546,8 @@ export function QueryBuilder({
                   getQueryFieldRecords={getQueryFieldRecords}
                   queryResource={queryResource}
                   recordSetId={recordSet?.id}
+                  results={resultsRef.current}
```

Comment:
Remember that changes to a ref don't trigger a re-render?
Thus, whatever prop value you give it, it might become stale.

Take a look at the `<QueryExportButton>` component and you will see that it only needs query results when the export button is clicked - it is not used during render, thus it doesn't need to be a state - that's exactly the reason why were able to convert from using state to a ref

<details><summary>Hint</summary>
<p>

Remember that you mutated the resultsRef object directly (by doing `resultsRef.current = ...`.
Thus, the object is mutated, but it's still the same object - the reference to an object remains unchanged, only the contents were updated - which means, you can pass the reference, not the value of the reference, and access the value of the reference only when you need it - when the button is clicked

</p>
</details> 

<details><summary>Solution</summary>
<p>

Replace
```ts
results={resultsRef.current}
```

with
```ts
resultsRef={resultsRef}
```

and fix the resulting type errors

</p>
</details>